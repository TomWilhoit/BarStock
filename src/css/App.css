/********************
FONTS
font-family: 'Roboto Condensed', sans-serif;
font-family: 'Open Sans Condensed', sans-serif;
font-family: 'Abel', sans-serif;
font-family: 'Open Sans', sans-serif;

COLORS
Orange #fdba35
Blue Teal #3ebab4
Medium Gray #5e666c

NEW PALETTE: 
Dark Gold: #EEAD0E
Light Gold: #F8DE9E
Dark Gray: #bbbbbb
Light Gray: #cfcfcf
Brown: #7d6c54
Black: #1d2023
White: #ffffff

*********************/

/* #region Global */
html {
  height: 100%;
  background: #dedede;
  /* background: #1d2023; */
}

body {
  height: 100%;
}

/* #endRegion Global */



.Header-container {
  padding-bottom: 2em;
}

.Footer-container {
  position: relative;
  width: 100%;
  text-align: center;
  /* background: #1d2023; */
}

