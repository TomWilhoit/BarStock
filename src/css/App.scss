/********************
FONTS
font-family: 'Roboto Condensed', sans-serif;
font-family: 'Open Sans Condensed', sans-serif;
font-family: 'Abel', sans-serif;
font-family: 'Open Sans', sans-serif;

COLORS
Orange #fdba35
Blue Teal #3ebab4
Medium Gray #5e666c

NEW PALETTE:
Dark Gold: #EEAD0E
Light Gold: #F8DE9E
Dark Gray: #bbbbbb
Light Gray: #cfcfcf
Brown: #7d6c54
Black: #1d2023
White: #ffffff

*********************/

/* #region Global */
html {
  height: 100%;
  background: #dedede;
  background: url(./dust_scratches.png);
}

body {
  height: 100%;
}

/* #endRegion Global */

#root {
  height: 100%;
}

.App {
  height: 100%;
}

.All-Content {
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}


.Header-container {
  margin-bottom: 2em;
}


/* #region Content Container  */


.content-container {
  display: table;
  max-width: 1200px;
  width: 100%;
  margin: auto;
}

.Inventory-container {
  display: table-cell;
  width: 60%;
  padding: 0em 0em 0em 2em;
}

.Totals-container {
  display: table-cell;
  width: 40%;
  padding: 0em 2em 0em 2em;
  vertical-align: bottom;
}

/* #endRegion Content Container */


.Submit-order {
  button {
    width: 100%;
    border-radius: 4px;
    outline: none;
    background: #EEAD0E;
    color: #fff;
    text-transform: uppercase;
    font-weight: 100;
    font-size: 1em;
    padding: .2em;
    font-family: 'Abel', sans-serif;
    cursor: pointer;
    &.disabled {
      pointer-events: none;
      opacity: .2;
    }
    &.enabled {
      pointer-events: all;
      opacity: 1;
    }
  }
}


.Footer-container {
  width: 100%;
  text-align: center;
  margin-top: 2em;
}



@media only screen and (max-width: 800px) {
  .content-container {
    display: block;
  }
  .Inventory-container {
    margin: auto;
    display: block;
    width: 100%;
    max-width: 600px;
    padding: 0;
    padding-bottom: 2em;
  }
  .Totals-container {
    display: block;
    margin: auto;
    width: 100%;
    padding: 0;
    max-width: 600px;
  }
}
