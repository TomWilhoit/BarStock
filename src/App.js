import React, { Component } from 'react';
import MockData from './mockData.js'
import './css/App.css';
import Inventory from './Inventory.js';
import Totals from './Totals.js';
import Header from './Header.js';
import Footer from './Footer.js';
import Login from './Login.js';



class App extends Component {
  constructor() {
    super();
    this.state = {
      allProducts: [MockData.distributor[0].inventory], 
      allMenu: [MockData.menu],
      cartItems: [],
      totalCost: 0,
      totalProjected: 0,
      currentUser: ""
    }
  }
  toggleAlert() {
    alert('toggle');
  }
  render() {
    return (
      <div className="App">
        {/* <div className="Login-container">
          <Login/>
        </div> */}
        <div className="Header-container">
          <Header/>
        </div>
        <div className="Inventory-container">
          <Inventory allProducts={this.state.allProducts}/>
        </div>
        <div className="Totals-container">
          <Totals allProducts={this.state.allProducts} 
                  allMenu={this.state.allMenu} />
        </div>
        <div className="Footer-container">
          <Footer/>
        </div>
      </div>
    )
  }
}

export default App;
