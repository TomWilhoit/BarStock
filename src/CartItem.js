import React, { Component } from 'react';
import App from './App.js';
import InventoryProd from './InventoryProd.js';

class CartItem extends Component {
    constructor(props) {
        super(props);
        this.state = {
            quantity: 0
        }
    }

    addProduct = event => {
        this.props.changeCart(this.props.product, "plusOne");
        const increaseQuantity = {
            quantity: this.state.quantity + 1
        }
        this.setState(increaseQuantity)
    }

    minusProduct = event => {
        this.props.changeCart(this.props.product, "minusOne");
        const decreaseQuantity = {
            quantity: this.state.quantity - 1
        }
        this.setState(decreaseQuantity)
    }

    render() {
        return(
            <div className="Single-product" id={this.props.product.inventory_code}>
                <span>{this.props.product.product}</span>
                <span>${this.props.product.price}</span>
                <span>
                    <button onClick={this.minusProduct}><i className="fas fa-minus-square"></i></button>
                    <span> {this.state.quantity} </span> 
                    <button onClick={this.addProduct}><i className="fas fa-plus-square"></i></button>
                </span>
            </div>
        ) 

    }
}

export default CartItem;