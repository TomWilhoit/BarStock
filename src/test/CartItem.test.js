import React from 'react';
import App from '../App';
import { shallow } from 'enzyme';
import CartItem from '../CartItem';


describe('CartItem', () => {
  let product = {
    "product": "Fireball",
    "inventory_code": 10001,
    "type": "liquor",
    "category": "whiskey",
    "price": 18.78,
    "size": 33.8,
    "unit": "ounces"
  }

  let wrapper;
  let inventory_code = product.inventory_code;
  let cartItems = [];

  beforeEach(() => {
    wrapper = shallow(
      <CartItem product={product} key={inventory_code} className="CartItem" changeCart={changeCart} cartItems={cartItems} />
        )})

  const changeCart = jest.fn();
  const currQuantity = jest.fn();

  it('should match snapshot when all data is passed correctly', () => {
    let cartItems = [{
      "product": "Fireball",
      "inventory_code": 10001,
      "type": "liquor",
      "category": "whiskey",
      "price": 18.78,
      "size": 33.8,
      "unit": "ounces"
    },
    {
      "product": "Jim Beam",
      "inventory_code": 10002,
      "type": "liquor",
      "category": "whiskey",
      "price": 16.99,
      "size": 25.3,
      "unit": "ounces"
    }]

    expect(wrapper).toMatchSnapshot();
  });

  it('should have a proper default state', () => {
    let cartItems = [
    {
      "product": "Jim Beam",
      "inventory_code": 10002,
      "type": "liquor",
      "category": "whiskey",
      "price": 16.99,
      "size": 25.3,
      "unit": "ounces"
    }]

    wrapper = shallow(
      <CartItem product={product} className="CartItem" changeCart={changeCart} cartItems={cartItems} />
        )

    expect(wrapper.state('quantity')).toEqual(0)
    });


    
})