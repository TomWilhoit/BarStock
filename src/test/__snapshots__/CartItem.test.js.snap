// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem should match snapshot when all data is passed correctly 1`] = `
<section
  className="Single-product"
  id={10001}
>
  <div
    className="product-img"
  >
    <img
      src="10001.png"
    />
  </div>
  <div
    className="product-info"
  >
    <h4>
      Fireball
    </h4>
    <h5>
      Price Per: 
      <span>
        $
        18.78
      </span>
    </h5>
  </div>
  <div
    className="product-quantity"
  >
    <section
      className="quantity-amount"
    >
      <button
        onClick={[Function]}
      >
        <i
          className="fas fa-minus"
        />
      </button>
      <div
        className="amount"
      >
        0
      </div>
      <button
        onClick={[Function]}
      >
        <i
          className="fas fa-plus"
        />
      </button>
    </section>
    <section
      className="quantity-label"
    >
      <p>
        Current Quantity
      </p>
    </section>
  </div>
</section>
`;
